doc/code_style_guidelines.md

```diff
@@ -0,0 +1,218 @@
+} while (0)
+```
+
+
+# Bracing and indentation
+
+## `if` statements
+
+This rule has already been explicitly stated in the [Linux kernel code style](https://www.kernel.org/doc/html/latest/process/coding-style.html) from which this code style inherits, but it will be repeated here:
+
+<span style="color: red">**Putting the body of an `if` statement on the same line as the condition is strictly prohibited.**</span>
+
+Example:
+```cpp
+if (foobar < 3) foobar = 45;      // Bad
+(foobar < 3 && (foobar = 45));    // Bad
+```
+
+Violating this rule will result in **instant rejection**.
+
+Examples of good if statement wordings:
+```cpp
+if (foobar < 3)
+	foobar = 45;
+
+if (foobar < 6) {
+	foobar = 62;
+	return;
+}
+```
+
+## Nested for loop exception
+
+Special exception to the standard bracing/indent rules for nested loops:
+If a nested loop iterates over a set of coordinates, it is permitted to omit the braces for all but the innermost loop and keep the outer loops at the same indentation level, like so:
+```cpp
+for (s16 z = pmin.Z; z <= pmax.Z, z++)
+for (s16 y = pmin.Y; y <= pmax.Y; y++)
+for (s16 x = pmin.X; x <= pmax.X; x++) {
+	// ... do stuff here ...
+}
+```
+
+
+# Do not be too C++y
+
+* Avoid passing non-`const` references to functions.
+* Don't use initializer lists unless absolutely necessary (initializing an object inside a class, or initializing a reference).
+* Try to minimize the use of exceptions.
+* Avoid operator overloading like the plague.
+* Avoid templates unless they are very convenient.
+* Usage of macros is not discouraged, just don't overdo it [like X.org](http://cgit.freedesktop.org/xorg/xserver/tree/randr/rrscreen.c?id=01e18af17f8dc91451fbd0902049045afd1cea7e#n325). It's better to use inline functions or lambdas instead.
+
+
+# Classes
+
+* **Class names are *PascalCase*, method names are *camelCase*.**
+* Don't put actual code in header files, unless it's a 4-liner, an inline function, or part of a template.
+* Class definitions should go in header files.
+* Substantial methods (over 4 lines) should be defined outside of the class definition.
+* Functions not part of any class should use `lowercase_underscore_style()`.
+
+
+# Comments
+
+* Doxygen comments are acceptable, but **please** put them in the header file.
+* Don't make uninformative comments like this:
+
+```cpp
+// Draw "Loading" screen
+draw_load_screen(L"Loading...", driver, font);
+```
+
+* Add comments to explain a non-trivial but important detail about the code, or explain behavior that is not obvious.
+* For comments with text, be sure to add a space between the text and the comment tokens:
+
+```cpp
+DoThingHere();  // This does thing    <--- yes!
+DoThingHere();  /* This does thing */ <--- yes!
+
+DoThingHere();  //This does thing      <--- no!
+DoThingHere();  /*This does thing*/    <--- no!
+DoThingHere();//This does thing        <--- no!
+```
+
+
+# Use STL, avoid Irrlicht containers, and no, Boost will not even be considered, so forget it
```

Comment:
Awfully long for a heading

Reply:
Found like this in the wiki.
Can't change easily without changing content.